tests/unit/test_data_preparation.py

```diff
@@ -0,0 +1,276 @@
+    INPUT_DATASET_LENGTH = 20
+
+    @staticmethod
+    def get_feature_schema() -> t.List:
+        return [
+            "ENSG00000162639",
+            "ENSG00000163568",
+            "ENSG00000243701",
+            "ENSG00000152580",
+            "ENSG00000253811",
+            "ENSG00000186439",
+            "ENSG00000254756",
+            "ENSG00000139211",
+            "ENSG00000185615",
+            "ENSG00000246379",
+            "ENSG00000266469",
+            "ENSG00000287710",
+            "ENSG00000262006",
+        ]
+
+    @staticmethod
+    def get_smaller_feature_schema() -> t.List:
+        return [
+            "ENSG00000162639",
+            "ENSG00000163568",
+            "ENSG00000243701",
+            "ENSG00000253811",
+            "ENSG00000186439",
+            "ENSG00000254756",
+            "ENSG00000139211",
+            "ENSG00000185615",
+            "ENSG00000246379",
+        ]
+
+    @staticmethod
+    def get_bigger_feature_schema() -> t.List:
+        return [
+            "ENSG00000162639",
+            "ENSG00000163568",
+            "ENSG00000243701",
+            "ENSG00000152580",
+            "ENSG00000253811",
+            "ENSG00000186439",
+            "ENSG00000254756",
+            "ENSG00000139211",
+            "ENSG00000185615",
+            "ENSG00000263674",
+            "ENSG00000246379",
+            "ENSG00000266469",
+            "ENSG00000287710",
+            "ENSG00000185869",
+            "ENSG00000262006",
+            "ENSG00000167113",
+        ]
+
+    @staticmethod
+    def get_schema_that_has_missing_and_extra() -> t.List:
+        return [
+            "ENSG00000163568",
+            "ENSG00000243701",
+            "ENSG00000152580",
+            "ENSG00000253811",
+            "ENSG00000186439",
+            "ENSG00000254756",
+            "ENSG00000139211",
+            "ENSG00000263674",
+            "ENSG00000246379",
+            "ENSG00000266469",
+            "ENSG00000287710",
+            "ENSG00000167113",
+        ]
+
+    @staticmethod
+    def get_same_schema_different_order() -> t.List:
+        return [
+            "ENSG00000162639",
+            "ENSG00000243701",
+            "ENSG00000163568",
+            "ENSG00000152580",
+            "ENSG00000186439",
+            "ENSG00000253811",
+            "ENSG00000254756",
+            "ENSG00000139211",
+            "ENSG00000246379",
+            "ENSG00000185615",
+            "ENSG00000266469",
+            "ENSG00000262006",
+            "ENSG00000287710",
+        ]
+
+    def test_data_validation_input_has_missing_features(self):
+        cas_feature_schema_list = self.get_feature_schema()
+        adata_feature_schema_list = self.get_smaller_feature_schema()
+        data_validation_error_was_raised = False
+        number_of_missing_features = 0
+        number_of_extra_features = 0
+        d = np.random.randint(0, 500, size=(self.INPUT_DATASET_LENGTH, len(adata_feature_schema_list)))
```

Comment:
It's always best to use private seeded random number generators (`np.random.RandomState(seed=...)`) in unit tests rather than the global `np.random`.